internal/api/schemas/v1.yaml

```diff
@@ -11,17 +11,168 @@ info:
+      parameters:
+      - name: namespaces
+        in: query
+        description: The namespaces of gateways to list
+        schema:
+          type: string
+      responses:
+        '200':
+          description: gateway response
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/GatewayPage"
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Error"
+    post:
+      description: 'Creates a new gateway.'
+      operationId: createGateway
+      requestBody:
+        description: Gateway to create
+        required: true
+        content:
+          application/json:
+            schema:
+              "$ref": "#/components/schemas/Gateway"
+      responses:
+        '201':
+          description: gateway response
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Gateway"
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Error"
+  "/namespaces/{namespace}/gateways/{name}":
+    get:
+      description: Returns the gateway based off of its unique name.
+      operationId: getNamespacedGateway
+      parameters:
+      - name: namespace
+        in: path
+        description: Namespace of gateway to fetch
+        required: true
+        schema:
+          type: string
+      - name: name
+        in: path
+        description: Name of gateway to fetch
+        required: true
+        schema:
+          type: string
+      responses:
+        '200':
+          description: gateway response
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Gateway"
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Error"
+    delete:
+      description: deletes a single gateway based on the Name supplied
+      operationId: deleteNamespacedGateway
+      parameters:
+      - name: namespace
+        in: path
+        description: Namespace of gateway to fetch
+        required: true
+        schema:
+          type: string
+      - name: name
+        in: path
+        description: Name of gateway to delete
+        required: true
+        schema:
+          type: string
+      responses:
+        '204':
+          description: gateway deleted
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Error"
+  "/gateways/{name}":
+    get:
+      description: Returns a gateway in the default namespace based off of its unique name.
```

Comment:
IIRC, mardown is valid here. I would do `"default"` just to indicate that we are talking about a specific, non-configurable namespace named "default"